tests/tests.h

```diff
@@ -182,6 +182,29 @@ void sort_file_contents (const std::string &filename)
 }
 
 
+/*
+ * simple ADLER32 checksum for a range of chars
+ */
+template <class IT>
+unsigned int checksum(const IT &begin, const IT &end)
+{
+  AssertThrow(sizeof(unsigned int)==4, ExcInternalError());
+
+  unsigned int a = 1;
+  unsigned int b = 0;
+
+  IT it = begin;
+
+  while (it != end)
+    {
+      a = (a + (unsigned char)*it) % 65521;
```

Comment:
This invites mistakes where someone passes in an iterator that returns something wider than a single char. How about iterating over the bytes returned by *it?